src/query/service/src/pipelines/processors/transforms/hash_join/row.rs

```diff
@@ -25,19 +25,9 @@ use parking_lot::RwLock;
 
 use crate::sessions::QueryContext;
 
-pub struct Chunk {
-    pub data_block: DataBlock,
-}
-
-impl Chunk {
-    pub fn num_rows(&self) -> usize {
-        self.data_block.num_rows()
-    }
-}
-
 pub struct RowSpace {
```

Comment:
It feels like this could be changed: `build_schema` is moved out. Then inside HashJoinState is `Arc<RwLock<RowSpace>>`, then `buffer` and `buffer_row_size` don't need locks, and `write_lock` doesn't need one. `gather `is not used as a method for `RowSpace`.